src/data/resolvers/user.ts

```diff
@@ -210,6 +216,79 @@ export const userResolvers = ({
       );
       return reputation.toString();
     },
+    async userReputationForTopDomains(_, { address, colonyAddress }) {
+      try {
+        const colonyClient = await colonyManager.getClient(
+          ClientType.ColonyClient,
+          colonyAddress,
+        );
+
+        // eslint-disable-next-line max-len
+        const userReputationForAllDomains: {
+          reputationAmount?: BigNumber;
+          domainId: number;
+          skillId: number;
+        }[] = await colonyClient.getReputationAcrossDomains(address);
+
+        // Extract only the relevant data and
+        // transform the user reputation amount on each domain to percentage
+        const formattedUserReputations = userReputationForAllDomains.map(
+          async (userReputation) => {
+            const totalColonyReputation = await getUserReputation(
+              colonyManager,
+              AddressZero,
+              colonyAddress,
+              userReputation.domainId,
+            );
+            const reputationPercentage = calculatePercentageReputation(
+              DECIMAL_PLACES,
+              userReputation.reputationAmount?.toString(),
+              totalColonyReputation.toString(),
+            );
+            return {
+              reputationPercentage,
+              domainId: userReputation.domainId,
+            };
+          },
+        );
+        const formattedUserReputationsResult = await Promise.all(
+          formattedUserReputations,
+        );
+
+        // Filter out the reputation percentages that are 0
+        const filteredUserReputations = formattedUserReputationsResult.filter(
+          (userReputation) =>
+            userReputation.reputationPercentage &&
+            userReputation.reputationPercentage !== ZeroValue.Zero,
+        );
+
+        // Sort out the percentages from highest to lowest
+        // and extract up to 3 reputations from the array
+        const topUserReputations = [...filteredUserReputations]
+          .sort((reputationA, reputationB) => {
+            const safeReputationA = new Decimal(
+              reputationA.reputationPercentage || 0,
+            );
+            const safeReputationB = new Decimal(
+              reputationB.reputationPercentage || 0,
+            );
+
+            if (safeReputationB.eq(safeReputationA)) {
+              return 0;
+            }
+            if (safeReputationB.lt(safeReputationA)) {
+              return -1;
+            }
+            return 1;
+          })
+          .slice(0, 3);
```

Comment:
So I assume this requirement is coming from product, because personally, I would argue that `Root` should always be sorted at the top, regardless of the user's reputation in it 

![Screenshot from 2021-09-10 15-26-02](https://user-images.githubusercontent.com/1193222/132853249-64f0a432-46f4-41f2-9b4c-9baf54927899.png)
![Screenshot from 2021-09-10 15-26-09](https://user-images.githubusercontent.com/1193222/132853254-d6a0fefa-321b-4716-b0b4-54b767992acf.png)

Reply:
Hmmm actually, Product didn't explicitly mention that Root needed to be in order with the rest nor on top of the list. 

In the design, the Root is on top but they seem to be arranged there from the lowest to the highest rep (with the root domain having the lowest percentage) or ordered from the first domain to the last domain.

I think I'm gonna ask Product about this

Reply:
So, according to Product, it would be good to keep Root at the top so I'm gonna change that :)